libsolidity/analysis/ContractLevelChecker.cpp

```diff
@@ -33,6 +33,38 @@ using namespace dev;
 using namespace langutil;
 using namespace dev::solidity;
 
+namespace {
+
+string getSignature(FunctionDefinition const& _def)
+{
+	return _def.name() + TypeProvider::function(_def)->toString(true);
+}
+
+template <class T>
+bool hasEqualNameAndParameters(T const& _a, T const& _b)
+{
+	return _a.name() == _b.name() &&
+		FunctionType(_a).asCallableFunction(false)->hasEqualParameterTypes(
+			*FunctionType(_b).asCallableFunction(false)
+	);
+}
+
+map<string, ContractLevelChecker::FunctionInfo> getOverridableFunctions(ContractDefinition const* _contract)
+{
+	map<string, ContractLevelChecker::FunctionInfo> _functions;
+
+	for (FunctionDefinition const* f: _contract->definedFunctions())
+		if (!f->name().empty())
```

Comment:
Please use `isFallback()` and `isConstructor()` and maybe even create a new helper function for that inside FunctionDefinition.